cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/KafkaLogAppender.java

```diff
@@ -40,11 +42,11 @@
   private final AtomicBoolean stopped = new AtomicBoolean(false);
 
   @Inject
-  public KafkaLogAppender(CConfiguration configuration) {
+  public KafkaLogAppender(CConfiguration configuration, @Named(Constants.Logging.KAFKA_TOPIC_PREFIX) String kafkaTopic) {
     setName(APPENDER_NAME);
     addInfo("Initializing KafkaLogAppender...");
 
-    this.producer = new SimpleKafkaProducer(configuration);
+    this.producer = new SimpleKafkaProducer(configuration, kafkaTopic);
```

Comment:
so can we read the kafka topic directly from the provided ``cConf`` object? Instead of injecting it in the constructor? Or did I miss something?